kernel/src/engine/arrow_utils.rs

```diff
@@ -900,18 +900,20 @@ fn parse_json_impl(json_strings: &StringArray, schema: ArrowSchemaRef) -> DeltaR
         return Ok(RecordBatch::new_empty(schema));
     }
 
-    // Use batch size of 1 to force one record per string input
-    let mut decoder = ReaderBuilder::new(schema.clone())
-        .with_batch_size(1)
-        .build_decoder()?;
+    let mut decoder = ReaderBuilder::new(schema.clone()).build_decoder()?;
     let parse_one = |json_string: Option<&str>| -> DeltaResult<RecordBatch> {
         let mut reader = BufReader::new(json_string.unwrap_or("{}").as_bytes());
-        let buf = reader.fill_buf()?;
-        let read = buf.len();
-        require!(
-            decoder.decode(buf)? == read,
-            Error::missing_data("Incomplete JSON string")
-        );
```

Comment:
Do we need to rescue this validation to make sure we are not working on an incomplete record in our decoder?

I think @scovich's upstream changes landed that should make this check easier?